substrate/extrinsic-pool/src/pool.rs

```diff
@@ -111,7 +111,7 @@ impl<Hash, VEx, S, E> Pool<Hash, VEx, S, E> where
 		let mut pool = self.pool.write();
 		let mut results = Vec::with_capacity(hashes.len());
 		for hash in hashes {
-			results.push(pool.remove(hash, is_valid));
+			results.push(pool.remove(hash, !is_valid));
```

Comment:
The `transaction-pool` API expects an inverted bool (`is_invalid`) - that was a bug that was causing the transactions to be logged as "canceled" instead of "invalid"

Reply:
Oh, maybe we should rename it then, since it's actually named `is_valid`? This is actually where my confusion comes from : )

https://github.com/paritytech/polkadot/blob/e89b8aa7e4ceb2b8f1348e027d61ebb9c820eec4/polkadot/transaction-pool/src/lib.rs#L389

Reply:
So the composition looks like this:
```
polkadot-transaction-pool -> substrate-extrinsic-pool -> [parity-]transaction-pool
```
The code in polkadot transaction pool is fine it uses `is_valid` and passes it further to extrinsics pool. Extrinsics pool uses the external `transaction-pool` (take from ethereum/parity) and that one is using `is_invalid` (https://docs.rs/transaction-pool/1.12.1/transaction_pool/struct.Pool.html#method.remove) hence the inversion here.